NitroxClient/Communication/Packets/Processors/PlayerTeleportedProcessor.cs

```diff
@@ -30,17 +30,28 @@ public override void Process(PlayerTeleported packet)
 
             Player.main.SetCurrentSub(subRoot);
         }
-        
-        Player.main.SetPosition(packet.DestinationTo.ToUnity());
-        // Freeze the player while he's loading its new position
-        Player.main.cinematicModeActive = true;
-        try
+
+        Vehicle currentVehicle = Player.main.currentMountedVehicle;
+        // Check to make sure the player is in a vehicle
+        if (currentVehicle != null)
```

Comment:
It's heavily advised to not check unity components and gameobjects against null. Instead just handle the object as a bool. 
```suggestion
        if (currentVehicle)
```
See: https://discussions.unity.com/t/how-can-i-check-if-an-object-is-null/23224/4